app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
+      vm.model = angular.copy(vm.modelCopy );
+      vm.angularForm.$setUntouched(true);
+      vm.angularForm.$setPristine(true);
+      miqService.miqFlash('warn', __('All changes have been reset'));
+    };
+
+    vm.cancelClicked = function() {
+      if (vm.newRecord) {
```

Comment:
As stated above, I prefer not to have common logic in the forms...common logic that we would have to keep repeating for every form. 
Time to cut it loose from the main form and have it in a separate component instead.